### PR TITLE
Support OCaml 4.08

### PR DESCRIPTION
**Pull request by anmonteiro**
This changeset adds support for OCaml 4.08.

The PR is very close to its final form, but currently adds 2 resolutions, one for `merlin-extend` with https://github.com/let-def/merlin-extend/pull/9 and one for Reason with https://github.com/facebook/reason/pull/2426

We may need to wait for releases for the above 2 packages before considering merging  or releasing this.

I'd love to have some folks test this out on their OCaml 4.08 projects. If you're currently using RLS and wanna try to use it with OCaml 4.08, please ping me directly if you need help building the project!

### CODE REVIEW

File: process_ocaml/408/ProcessExtra.re [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+    } else {
+      List.hd(Collector.scopeExtent^);
+    }
+  };
+  let addScopeExtent = loc => Collector.scopeExtent := [loc, ...Collector.scopeExtent^];
+  let popScopeExtent = () => if (List.length(Collector.scopeExtent^) > 1) {
+    Collector.scopeExtent := List.tl(Collector.scopeExtent^);
+  };
+
+  let rec addForLongident = (top, path: Path.t, txt: Longident.t, loc) => {
```
Comment (jchavarri) [inline process_ocaml/408/ProcessExtra.re]: @anmonteiro There were some recent changes merged in `master` that were added to this file: https://github.com/jaredly/reason-language-server/pull/303/files#diff-7d9b7cbbb32d6528ce43ccd07615acfeR261.

Please let me know if you want me to add them, it should be just a copy-paste matter :)

Comment (anmonteiro) [reply process_ocaml/408/ProcessExtra.re]: Thanks @jchavarri. I'll look at that when I get around to fix the merge conflicts.

**Comment by josselinauguste** [comment]
does anything block this PR?

**Comment by anmonteiro** [comment]
@josselinauguste this is currently waiting for an upstream release for Reason as well as Jared's time which is a little limited right now.

**Comment by jaredly** [comment]
@anmonteiro how's the status on this? Is it ready for review?

**Comment by anmonteiro** [comment]
@jaredly yeah I think so. The only thing that might be worth doing is updating the OCaml sources to the newly released 4.08.1

**Comment by Lupus** [comment]
I've built extension from this branch and tried it with OCaml 4.08 (opam switch), seems to be working fine! Would be great to have this released.

**Comment by anmonteiro** [comment]
@Lupus did you try this with OCaml 4.08.1?

**Comment by anmonteiro** [comment]
Alright, I just rebased this on current master and built it against OCaml 4.08.1.

@jaredly This should now be ready to go

**Comment by Lupus** [comment]
> @Lupus did you try this with OCaml 4.08.1?

`ocaml-base-compiler.4.08.1` is what reported by `opam switch`.

**Comment by jaredly** [comment]
ok, given that CI is happy, I'll merge this in

**Comment by jaredly** [comment]
thanks for doing so much hard work on this!

**Pull request merged**